API_GUIDE.md

```diff
@@ -45,23 +45,26 @@ for data, target in train_loader:
   loss = loss_fn(output, target)
   loss.backward()
   xm.optimizer_step(optimizer)
+  xm.mark_step()
 ```
 
 The above is only running on one TPU core though, and the time spent to send data to device is serial/inline with the TPU computation.
 For simple experiments, or for inference tasks which are not latency-sensitive it might be still OK, but the following methods allow for better scalability.
 
-Note the `xm.optimizer_step(optimizer)` line which replaces the usual `optimizer.step()`. This is required because of the way XLA tensors work: operations are not executed immediately, but rather added to a graph of pending operations which is only executed when its results are required. Using `xm.optimizer_step(optimizer)` acts as an execution barrier which forces the evaluation of the graph accumulated for a single step. Without this barrier, the graph would only be evaluated when evaluating the accuracy of the model, which is only done at the end of an epoch, for this example. Even for small models, the accumulated graph would be too big to evaluate at the end of an entire epoch.
+Note the `xm.mark_step()` line. This is required because of the way XLA tensors work: operations are not executed immediately, but rather added to a graph of pending operations which is only executed when its results are required. Using `xm.mark_step()` acts as an execution barrier which forces the evaluation of the graph accumulated for a single step. Without this barrier, the graph would only be evaluated when evaluating the accuracy of the model, which is only done at the end of an epoch, for this example. Even for small models, the accumulated graph would be too big to evaluate at the end of an entire epoch.
```

Comment:
barrier is now in mark_step

Reply:
No, no. This is wrong.
The comment is about optimizer step.
They have to understand that that is needed.
Let's leave this as is for now.
For them, whether the barrier happen in xm.optimizer_step() or later in the data loader, should not matter.
Eventually we can have two separate comments.